File: 2026POTD/JANUARY/03-01-2026/1411.swift
class Solution {
    func numOfWays(_ n: Int) -> Int {
        let MOD = 1_000_000_007
        var aba = 6
        var abc = 6
        if n == 1 {
            return (aba + abc) % MOD
        }
        for _ in 2...n {
            let newAba = (aba * 3 + abc * 2) % MOD
            let newAbc = (aba * 2 + abc * 2) % MOD
            aba = newAba
            abc = newAbc
        }
        return (aba + abc) % MOD
    }
}

File: 2026POTD/JANUARY/03-01-2026/1411.dart
class Solution {
  int numOfWays(int n) {
    const int MOD = 1000000007;
    int aba = 6, abc = 6;
    for (int i = 2; i <= n; i++) {
      int newAba = (aba * 3 + abc * 2) % MOD;
      int newAbc = (aba * 2 + abc * 2) % MOD;
      aba = newAba;
      abc = newAbc;
    }
    return (aba + abc) % MOD;
  }
}